### DISCUSSION TITLE
blog/repository-pattern-caching-hangfire-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#42] [General]
# blog/repository-pattern-caching-hangfire-aspnet-core/

In this article, we will learn about an approach with which we will build a SUPER-FAST Repository Implementation using all the latest libraries and concepts. We will be implementing Repository Pattern with Caching and Hangfire in ASP.NET Core along with Onion Architecture just to make sure our code is well organized and be readily used in any random project.

https://codewithmukesh.com/blog/repository-pattern-caching-hangfire-aspnet-core/

**Comment by AbusaVLM** [comment]
Great example for in-memory that I was able to get running even with the minimal API Program.cs file instead of the StartUp.cs mode.  A very nice tutorial.  Will have to do some digging for Redis and other caching methods.  Thanks so much for sharing and all resources you've made available to the development community.

**Comment by khalifa005** [comment]
using keyed services

    #region KeyedServices
    //[FromKeyedServices(CacheTechEnum.Memory)] ICache bigCache - inside services
    services.AddKeyedTransient<ICacheService, MemoryCacheService>(CacheTechEnum.Memory);
    services.AddKeyedTransient<ICacheService, RedisCacheService>(CacheTechEnum.Redis);
    #endregion


  private readonly AppDbContext _dbContext;
  private readonly static CacheTechEnum cacheTech = CacheTechEnum.Memory;
  private readonly string cacheKey = $"{typeof(T)}";
  private readonly ICacheService _cacheService;
  public GenericRepository(
    AppDbContext dbContext,
    [FromKeyedServices(CacheTechEnum.Memory)] ICacheService memoryCache)
  {
    _dbContext = dbContext;
    _cacheService = memoryCache;
  }
